### PR TITLE
Add product quality behaviours to mobile mastery

### PR DESCRIPTION
**Pull request by alexfish**
Mobile mastery is currently missing behaviours related to product quality, this pull request proposes some new behaviours that reflect building high quality products.

These changes could also translate well to the web framework or any engineers building customer facing products.

**For example:**

* Handling empty states.
* Working closely in collaboration with designers.
* Handling loading states.
* Gracefully handling errors.
* Proactively suggesting improvements.

### CODE REVIEW

**Comment by alexfish** [comment]
I'll find time later today to fix these build errors.

File: frameworks/engineering/mobile.md [hunk 1]
```diff
@@ -172,6 +172,7 @@ topics:
           - "Builds simple screens or flows"
           - "Asks questions and actions feedback from PRs and other engineers"
           - "Fixes simple bugs or issues"
+          - "Produces well-polished interfaces in collaboration with a designer"
```
Comment (andysmart) [inline frameworks/engineering/mobile.md]: `well-polished` might need more definition, it's a little up to interpretation of engineers + managers.

File: frameworks/engineering/mobile.md [hunk 2]
```diff
@@ -201,6 +203,7 @@ topics:
           - "Builds non-trivial, coordinated app flows with multiple services and data sources"
           - "Builds complex UI layouts in code or via IDE"
           - "Contributes to group standards, impactful in improving our processes"
+          - "Proactively improves empty, error and loading states in screens & flows they encounter, works closely with designers to identify problem areas and suggest improvements"
```
Comment (andysmart) [inline frameworks/engineering/mobile.md]: Love this one

File: frameworks/engineering/mobile.md [hunk 3]
```diff
@@ -212,6 +215,7 @@ topics:
           - "Writes code that serves as a definitive example for new engineers"
           - "Identifies and fixes security vulnerabilities"
           - "Sought out for reviewing complex changes or commenting on larger scale proposals"
+          - "Builds non trivial complex flows that serve as an example of a high quality product"
```
Comment (andysmart) [inline frameworks/engineering/mobile.md]: Would it be worth getting something around interaction design here? "high quality" could mean a lot of things, code quality, design etc?

**Comment by alexfish** [comment]
Agreed it's quite open to interpretation at the moment, especially levels 1-2 and what high quality means.

**Comment by alexfish** [comment]
I've booked in 30mins next week to pair on this.

File: frameworks/engineering/mobile.md [hunk 4]
```diff
@@ -172,6 +172,7 @@ topics:
           - "Builds simple screens or flows"
           - "Asks questions and actions feedback from PRs and other engineers"
           - "Fixes simple bugs or issues"
+          - "Handles empty, error & loading states in simple screens or flows"
```
Comment (andysmart) [inline frameworks/engineering/mobile.md]: Shall we remove this? Level 1 is pretty much covered IMO

File: frameworks/engineering/mobile.md [hunk 5]
```diff
@@ -186,6 +187,7 @@ topics:
           - "Appropriately uses common functional reactive programming concepts"
           - "Designs database models"
           - "Appropriately handles errors or failure conditions in their code"
+          - "Handles empty, error, and loading states gracefully in multi screen flows"
```
Comment (andysmart) [inline frameworks/engineering/mobile.md]: How about calling out designers specifically? "Works with designers to handle empty..."

File: frameworks/engineering/mobile.md [hunk 6]
```diff
@@ -212,6 +215,7 @@ topics:
           - "Writes code that serves as a definitive example for new engineers"
           - "Identifies and fixes security vulnerabilities"
           - "Sought out for reviewing complex changes or commenting on larger scale proposals"
+          - "Builds non trivial complex flows that serve as an example of high quality interaction design"
```
Comment (andysmart) [inline frameworks/engineering/mobile.md]: Shall we lose "non trivial complex", be more explicit?

File: frameworks/engineering/mobile.md [hunk 7]
```diff
@@ -231,6 +235,7 @@ topics:
         criteria:
           - "Anticipates large or significant platform changes, work with team to define possible solutions. Starts or finishes these projects."
           - "Takes high level requirements and turns them into discrete technical concerns"
+          - "Instills and maintains a culture of high quality in their discipline"
```
Comment (alexfish) [inline frameworks/engineering/mobile.md]: high quality interaction design

**Comment by mshipton** [comment]
Is it worthwhile mentioning behaviours relating to quality outside of coding?
- explores edge cases; ensures these are raised with the squad and handled appropriately
- collaborate with testers to ensure those cases are covered; supporting where required to put the app in to the correct state.

Something that says that quality doesn't often end with writing a solution. There is a degree of ownership/responsibility to ensure it's carried through to the feature is done?

**Comment by andysmart** [comment]
@mshipton yes, absolutely - I think these are great actually, perhaps you and Alex could pair on a PR to the framework to represent this? I think the intent behind this one was design lead but we should be making more changes to the framework than we are, as we learn.

**Pull request merged**